### ISSUE TITLE
the targets prop passed to child component is undefined

### ISSUE THREAD

**Issue by zachary822** [reactstrap/reactstrap#1686]
- components: `TooltipPopoverWrapper`
- reactstrap version `#8.1.1`
- import method `umd/csj/es` 
- react version `#16.10.2`
- bootstrap version `#4.3.1`

### What is happening?

In the [`updateTarget`](https://github.com/reactstrap/reactstrap/blob/45775c01da58e4e29d752367b43a730258398d78/src/TooltipPopoverWrapper.js#L296-L304) method. If `this.props.target` is a function that returns a single element (not array), then the [`Array.from(newTarget)`](https://github.com/reactstrap/reactstrap/blob/45775c01da58e4e29d752367b43a730258398d78/src/TooltipPopoverWrapper.js#L300) will return `[]` (empty array). This will cause `this._targets` to be set to an empty array. In turn [`this.currentTargetElement` ](https://github.com/reactstrap/reactstrap/blob/45775c01da58e4e29d752367b43a730258398d78/src/TooltipPopoverWrapper.js#L301) to be set to `undefined`. Which will cause `undefined` to be passed to `PopperContent` as its `target` prop. Resulting in no tooltips showing up. 

### What should be happening?

Tooltip should show up if `target` prop is valid.

### Steps to reproduce issue

1. passing a function to the `target` prop of `Tooltip`
2. ...

### Error message in console

```
backend.js:6 Warning: Failed prop type: The prop `target` is marked as required in `PopperContent`, but its value is `undefined`.
    in PopperContent (created by TooltipPopoverWrapper)
    in TooltipPopoverWrapper (created by Popover)
```

### Code

https://stackblitz.com/edit/reactstrap-uynews?embed=1&file=Example.js

**Comment by zachary822** [comment]
I made a pull request to a possible fix #1687

**Comment by bradchristensen** [comment]
@zachary822 I've noticed the issue you mentioned when trying to upgrade from Reactstrap 8.0.1 (last working version for tooltips using refs for me) to 8.2.0 as per the screenshot below.

<img src="https://user-images.githubusercontent.com/6006249/71350484-4e977800-25d6-11ea-94d4-8671a54cfac8.png" width="514.66" />

I agree that filtering out falsy values in the `getTarget` function seems like the way to fix this.
